downstairs/src/stats.rs

```diff
@@ -0,0 +1,196 @@
+// Copyright 2021 Oxide Computer Company
+use super::*;
+
+use dropshot::{ConfigDropshot, ConfigLogging, ConfigLoggingLevel};
+use omicron_common::api::internal::nexus::ProducerEndpoint;
+use oximeter::{
+    types::{Cumulative, Sample},
+    Error, Metric, Producer, Target,
+};
+use oximeter_producer::{Config, Server};
+
+// These structs are used to construct the required stats for Oximeter.
+#[derive(Debug, Copy, Clone, Target)]
+pub struct DsStat {
```

Comment:
I'd probably recommend renaming these structures to something more descriptive. I've been thinking of these as akin to DTrace providers and probes, in which case the names should be meaningful enough that the data they export could be easily interpreted. And ideally, one would be able to search for them without knowing their exact names. Something like `CrucibleDownstairs` for the target and `{Connect,Write,Read,Flush}Count` for the metrics might be a good choice.

Reply:
Names have been changed.